systemtests/src/test/java/io/enmasse/systemtest/TestBaseWithDefault.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2017 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.enmasse.systemtest;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.junit.runner.Result;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class TestBaseWithDefault extends TestBase {
+    private static final String defaultAddressTemplate = "-default-";
+    protected AddressSpace defaultAddressSpace = new AddressSpace(getAddressSpaceType().name().toLowerCase() + defaultAddressTemplate + "0", getAddressSpaceType());
+
+    protected abstract AddressSpaceType getAddressSpaceType();
+
+    @Rule
+    public TestWatcher watcher = new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+            Logging.log.info("test failed:" + description);
+            if (getSharedAddressSpace() != null) {
+                Logging.log.info("default address space '{}' will be removed", defaultAddressSpace);
+                try {
+                    deleteAddressSpace(defaultAddressSpace);
+                    initializeSharedAddressSpace();
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                }
+            }
+        }
+    };
+
+    @Override
+    public void testRunFinished(Result result) throws Exception {
+        try {
+            if (getSharedAddressSpace() != null) {
+                deleteAddressSpace(getSharedAddressSpace());
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public AddressSpace getSharedAddressSpace() {
+        return defaultAddressSpace;
+    }
+
+    @Before
+    public void setupDefault() throws Exception {
+        if (environment.isMultitenant()) {
```

Comment:
this method doesn't exist since your last commit

Reply:
Fixed :)